projects/common/src/time/time.ts

```diff
@@ -32,11 +32,27 @@ export class Time {
     return this._date;
   }
 
+  public static parse(label: string, isUTC: boolean = false): Time {
+    const time: string[] = label.split(' ');
+    const meridian: string = time.pop()!;
+    const [hours, minutes, seconds, milliseconds] = time[0].split(':').map(value => +value);
+
+    return new Time(Time.get24HrClockHours(hours, meridian), minutes, seconds, milliseconds, isUTC);
+  }
+
   public toISOString(): string {
     return this.date.toISOString().substring(11);
   }
 
   public equals(other?: Time): boolean {
     return this.toISOString() === other?.toISOString();
   }
+
+  private static get24HrClockHours(hours: number, meridian: string): number {
```

Comment:
meridian should be limited to `'AM' | 'PM'`. it'll be good to type it.

Reply:
updated